src/mbgl/style/sources/geojson_source.cpp

```diff
@@ -69,18 +70,30 @@ void GeoJSONSource::loadDescription(FileSource& fileSource) {
             observer->onSourceError(
                 *this, std::make_exception_ptr(std::runtime_error("unexpectedly empty GeoJSON")));
         } else {
-            conversion::Error error;
-            std::shared_ptr<GeoJSONData> geoJSONData;
-            if (optional<GeoJSON> geoJSON = conversion::convertJSON<GeoJSON>(*res.data, error)) {
-                geoJSONData = GeoJSONData::create(*geoJSON, impl().getOptions());
-            } else {
-                // Create an empty GeoJSON VT object to make sure we're not infinitely waiting for tiles to load.
-                Log::Error(Event::ParseStyle, "Failed to parse GeoJSON data: %s",
-                           error.message.c_str());
-            }
-            baseImpl = makeMutable<Impl>(impl(), std::move(geoJSONData));
-            loaded = true;
-            observer->onSourceLoaded(*this);
+            auto makeImplInBackground = [currentImpl = baseImpl, data = res.data]() -> Immutable<Source::Impl> {
```

Comment:
Is this safe? Because `baseImpl` is mutable.

Reply:
It is `Immutable`, pls see https://github.com/mapbox/mapbox-gl-native/blob/master/include/mbgl/style/source.hpp#L68